lib/sisimai/bite/email/facebook.rb

```diff
@@ -0,0 +1,274 @@
+                v['action'] = cv[1].downcase
+
+              elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+                # Status: 5.1.1
+                # Status:5.2.0
+                # Status: 5.1.0 (permanent failure)
+                v['status'] = cv[1]
+
+              elsif cv = e.match(/\A[Rr]emote-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Remote-MTA: DNS; mx.example.jp
+                v['rhost'] = cv[1].downcase
+
+              elsif cv = e.match(/\A[Ll]ast-[Aa]ttempt-[Dd]ate:[ ]*(.+)\z/)
+                # Last-Attempt-Date: Fri, 14 Feb 2014 12:30:08 -0500
+                v['date'] = cv[1]
+
+              else
+                if cv = e.match(/\A[Dd]iagnostic-[Cc]ode:[ ]*(.+?);[ ]*(.+)\z/)
+                  # Diagnostic-Code: smtp; 550 5.1.1 RCP-P2
+                  #     http://postmaster.facebook.com/response_codes?ip=192.0.2.135#rcp Refused due to recipient preferences
+                  v['spec'] = cv[1].upcase
+                  v['diagnosis'] = cv[2]
+
+                elsif p =~ /\A[Dd]iagnostic-[Cc]ode:[ ]*/ && cv = e.match(/\A[ \t]+(.+)\z/)
+                  # Continued line of the value of Diagnostic-Code header
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + cv[1]
+                  havepassed[-1] = 'Diagnostic-Code: ' + e
+                end
+              end
+
+            else
+              if cv = e.match(/\A[Rr]eporting-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Reporting-MTA: dns; mx.example.jp
+                next if connheader['lhost'].size > 0
+                connheader['lhost'] = cv[1].downcase
+                connvalues += 1
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                connheader['date'] = cv[1]
+                connvalues += 1
+              end
+
+            end
+          end
+        end
+        return nil if recipients.zero?
+        require 'sisimai/string'
+
+        dscontents.map do |e|
+          e['lhost']   ||= connheader['lhost']
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          if cv = e['diagnosis'].match(/\b([A-Z]{3})[-]([A-Z])(\d)\b/)
+            # Diagnostic-Code: smtp; 550 5.1.1 RCP-P2
+            lhs = cv[1]
+            rhs = cv[2]
+            num = cv[3]
+
+            fbresponse = sprintf('%s-%s%d', lhs, rhs, num)
+          end
+
+          catch :SESSION do
+            ReFailure.each_key do |r|
+              # Verify each regular expression of session errors
+              ReFailure[r].each do |rr|
+                # Check each regular expression
+                next unless fbresponse == rr
+                e['reason'] = r.to_s
+                throw :SESSION
+              end
+            end
+          end
+
+          unless e['reason']
```

Comment:
__[rubocop]__
* Style/Next: Use `next` to skip iteration. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->